src/Shared/RoslynUtils/SymbolExtensions.cs

```diff
@@ -74,6 +74,21 @@ public static bool HasAttribute(this ImmutableArray<AttributeData> attributes, I
         return false;
     }
 
+    public static bool TryGetAttribute(this ImmutableArray<AttributeData> attributes, INamedTypeSymbol attributeType, [NotNullWhen(true)] out AttributeData? matchedAttribute)
```

Comment:
nit: `public static bool HasAttribute(this ImmutableArray<AttributeData> attributes, INamedTypeSymbol attributeType)` implementation can be updated to call TryGetAttribute and return the bool and ignore the matched value.